### ISSUE TITLE
view compile with custom elements

### ISSUE THREAD

**Issue by caperavensoftware** [aurelia/templating#521]
I got view compliation to work fine with this code.

```
        let viewFactory =  this.viewCompiler.compile(this.html, this.resources);
        let view = viewFactory.create(this.container);
        let element = document.querySelector("#body");
        let viewSlot = new ViewSlot(element, true);
        viewSlot.add(view);
        viewSlot.attached();
        view.bind(this);
```

as long as the code contains standard html everything is fine, but as soon as you have a custom element, the require is not being processed.

A example of this:
```
<template><require from="pragma-tabsheet/pragma-tabsheet"></require><pragma-tabsheet><div data-tab="tab1"><div class="group"><h2>Group 1</h2></div></div><div data-tab="tab2"></div><div data-tab="tab3"></div></pragma-tabsheet></template>
```

If you copy this into the view it works fine, but as soon as you compile it from the string, it injects everything correctly but the custom element is not instantiated in code so none of the initialization takes place.

what must I do to get 
```
<require from="pragma-tabsheet/pragma-tabsheet"></require>
```
processed as part of the compilation process.

Others have asked the same question:
http://stackoverflow.com/questions/41468304/aurelia-how-to-register-view-resources-for-viewcompiler

**Comment by caperavensoftware** [comment]
I found that if I do the following I can get it to work but I don't want to do this.

1. Add the custom control to global resources
```
.globalResources('pragma-tabsheet/pragma-tabsheet');
```

2. Enhance the body element
```
this.templateEngine.enhance({element: element, bindingContext: this });
```

I would have thought that the compile and bind process of the view would have enhanced it for you.

I also found that if I try and enhance it with out having the global resource it also does not work.

am I missing a step in the process somewhere so that I don't have to do these two things?
